src/components/atomic-swap/exchange.tsx

```diff
@@ -114,15 +144,36 @@ export const Exchange = (props: ExchangeProps) => {
           </>
         );
       }
-      case 3: {
+      case 5: {
         const submit = async () => {
           const server = getServer(props.networkDetails);
 
           setIsSubmitting(true);
 
+          const tx = TransactionBuilder.fromXDR(
+            xdr.TransactionEnvelope.fromXDR(signedTx, "base64"),
+            props.networkDetails.networkPassphrase,
+          ) as Transaction<Memo<MemoType>, Operation[]>;
+
+          const txSim = await server.simulateTransaction(tx);
+          const preparedTransaction = assembleTransaction(
+            tx,
+            props.networkDetails.networkPassphrase,
+            txSim,
+            xdr.LedgerFootprint.fromXDR(
+              Buffer.from(originalFootprint, "base64"),
+            ),
```

Comment:
you can use the built-in decoder on these:
```suggestion
            xdr.LedgerFootprint.fromXDR(originalFootprint, "base64"),
```

Reply:
ah cool thanks